proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+import "redpanda/api/auth/v1/authorization.proto";
+
+// Defines the MCP server resource.
+message MCPServer {
+  option (google.api.resource) = {
+    type: "redpanda.api.dataplane.v1/MCPServer"
+    singular: "mcp_server"
+    plural: "mcp_servers"
+  };
+  // MCP Server ID.
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_/]+$"
+  ];
+
+  // User-friendly MCP server name.
+  string display_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+
+  // Optional MCP server description.
+  string description = 3 [(buf.validate.field).string.max_len = 256];
+
+  // A configuration file for the MCP server.
+  message Configuration {
+    // The path to the configuration file, this path must be relative and also valid in to layout in a filesystem,
+    // when considering many paths in a full configuration.
+    string path = 1 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true,
+      // Force a relative path pattern
+      (buf.validate.field).string.pattern = "^[A-Za-z0-9-_][A-Za-z0-9-_ /]+$",
+      (buf.validate.field).string.min_len = 3,
+      (buf.validate.field).string.max_len = 128
+    ];
+    // The contents of the configuration file in YAML format.
+    string contents = 2 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.min_len = 1
+    ];
+  }
+
+  // All the configuration files for the MCP server.
+  repeated Configuration config_files = 4 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).repeated.min_items = 1,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The Redpanda Connect MCP server configuration in files.."}
+  ];
+
+  // The number of resources that are guaranteed to be assigned to the MCP server.
+  Resources resources = 5 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "The requested amount of resources for the MCP server. Depending on the backend, the resources might be guaranteed, or might be tweaked based on the utilization of the MCP server."}];
+
+  // The current MCP server state.
+  State state = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  Status status = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // URL to connect to the MCP server
+  string url = 8 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Tags are key-value pairs that can be assigned to a MCP server resource.
+  // They help organize MCP servers and enable filtering when listing them.
+  map<string, string> tags = 9 [(buf.validate.field).map = {
```

Comment:
[nit]
call me esoteric, but: since this is a new resource, i would move this field "up" around id/display_name. it was only at such a high field index because it was added to pipeline: later on